test/algorand-lib/Test/Data/Algorand/Transaction/Examples.hs

```diff
@@ -70,16 +70,16 @@ unit_example_21_encoding =
     -- This was generated manually using Python SDK
     example_21_encoded :: ByteString
     example_21_encoded = result
-      "iqRhcGFhkcQEdGVzdKRhcGF0ksQgACoyATtqMON+4ohUJO59fQVV6uCTn7aa/GvfndL+5/7EIAAHBAuPYqMysOAF8ALIwKUWNGgBCjFYJ8bPUnx4aXnnpGFwZmGSzRWzzRoKpGFwaWRko2ZlZc0E0qJmds0jKKJnaMQgMf0h6zjkEIEZPtNM3zsrg+iHQFS0fZxhgr7w35I464OibHbNIzKjc25kxCAJ+9J2LAj4bFrmv23Xp6kB3mZ111Dgfoxcdphkfbbh/aR0eXBlpGFwcGw="
+      "i6RhcGFhkcQEdGVzdKRhcGFupGNhbGykYXBhdJLEIAAqMgE7ajDjfuKIVCTufX0FVergk5+2mvxr353S/uf+xCAABwQLj2KjMrDgBfACyMClFjRoAQoxWCfGz1J8eGl556RhcGZhks0Vs80aCqRhcGlkZKNmZWXNBNKiZnbNIyiiZ2jEIDH9Ies45BCBGT7TTN87K4Poh0BUtH2cYYK+8N+SOOuDomx2zSMyo3NuZMQgCfvSdiwI+Gxa5r9t16epAd5mdddQ4H6MXHaYZH224f2kdHlwZaRhcHBs"
```

Comment:
@rinn7e did you test that it's correct? Did you use Python SDK as written in the comment?

Reply:
@gromakovsky  I simply used the current function to generate this. I try to setup python sdk but it was not obvious how to generate this and it starts to be time-consuming. If you think it is necessary to do so, I'll try again.

Reply:
@rinn7e I am afraid that it may not work correctly. We are changing the binary format of data serialization. `OnComplete` is part of a transaction and there is a specific serialization format for transactions. In particular, this format is used for signing. Previously, in MsgPack (binary format) `OnComplete` was serialized as a number. But now it's serialized as text. As result, transaction serialization has changed and I think we won't be able to produce valid signatures for transactions with `OnComplete` values.
For example, you can see that `transactionId example_21` has changed, even though `example_21` hasn't changed. It's very suspicious.

Actually, since `OnComplete` has different format in MsgPack and JSON, I would change it to be an ADT with constructors for each possible value and define different serialization to JSON and MsgPack.

Reply:
> For example, you can see that transactionId example_21 has changed, even though example_21 hasn't changed. It's very suspicious.

That's because `onCompleteNoOp` is a string now. 

> I would change it to be an ADT
I check the file history and previously it was an ADT but somehow got changed to be just number. There may be a reason why it was no longer done as ADT or maybe it is just complicated thing.

I'll try to setup python sdk to test this.

Reply:
@gromakovsky So after installing python sdk with no success, I try the javascript sdk instead.

Managed to set it up fine, and I try to input our transaction data to the javascript sdk and notice that it has more fields: https://github.com/algorand/js-algorand-sdk/blob/d99ba0b96d373e6e51701b1c5ea8ab2cb3df06e5/tests/5.Transaction.js#L452-L478

Mainly `approvalProgram`, `clearProgram`, `numGlobalInts`, `numGlobalByteSlices`, `numLocalInts`, `numLocalByteSlices` and some other fields. Since we have different field like this it is quite tricky to know how to fill them. Hope to get your advice on this.